### PR TITLE
add integration test for pod with pvc has node-affinity to non-existent/illegal nodes

### PR DESCRIPTION
**Pull request by chengjoey**
#### What type of PR is this?
/kind cleanup
/sig scheduling

#### What this PR does / why we need it:
an integration test for pod with pvc has node-affinity to non-existent/illegal nodes could be schduable
related pr #124559
related issue #123465 

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: test/integration/scheduler/filters/filters_test.go [hunk 1]
```diff
@@ -2036,6 +2037,53 @@ func TestUnschedulablePodBecomesSchedulable(t *testing.T) {
...
+			name: "pod with pvc has node-affinity to non-existent/illegal nodes",
+			init: func(cs kubernetes.Interface, ns string) error {
+				storage := v1.VolumeResourceRequirements{Requests: v1.ResourceList{v1.ResourceStorage: resource.MustParse("1Mi")}}
+				volType := v1.HostPathDirectoryOrCreate
+				pv, err := testutils.CreatePV(cs, st.MakePersistentVolume().
+					Name("pv-has-non-existent-nodes").
+					AccessModes([]v1.PersistentVolumeAccessMode{v1.ReadWriteOncePod}).
+					Capacity(storage.Requests).
+					HostPathVolumeSource(&v1.HostPathVolumeSource{Path: "/tmp", Type: &volType}).
+					NodeAffinityIn("kubernetes.io/hostname", []string{"node-available", string(uuid.NewUUID()), string(uuid.NewUUID())}). // one node exist, two don't
```
Comment (AxeZhan) [inline test/integration/scheduler/filters/filters_test.go:2050]: Do we actually need to use UUID here? Won't a simple name like `non-existing` work as expected?

File: test/integration/scheduler/filters/filters_test.go [hunk 2]
```diff
@@ -2036,6 +2037,53 @@ func TestUnschedulablePodBecomesSchedulable(t *testing.T) {
 				return deletePod(cs, "pod-to-be-deleted", ns)
 			},
 		},
+		{
+			name: "pod with pvc has node-affinity to non-existent/illegal nodes",
```
Comment (AxeZhan) [inline test/integration/scheduler/filters/filters_test.go:2041]: Can we also add a case which pod got scheduled after node updated its label?

Maybe sth like this:

1. Pod with pv's affinity with `foo in bar` created
2. Node with label `foo = foo` created
3. Verify that the pod is not scheduled
4. Update the node with label `foo = bar`
5. Verify that the pod is scheduled successfully.

**Comment by alculquicondor** [comment]
/approve
I'll leave LGTM to @AxeZhan

File: test/integration/scheduler/filters/filters_test.go (hunk 1 above)
Comment (chengjoey) [reply test/integration/scheduler/filters/filters_test.go]: done

File: pkg/scheduler/testing/wrappers.go [hunk 3]
```diff
@@ -882,6 +882,20 @@ func (p *PersistentVolumeWrapper) HostPathVolumeSource(src *v1.HostPathVolumeSou
 	return p
 }
 
+// NodeAffinityIn creates a HARD node affinity (with the operator In)
+// // and injects into the pv.
```
Comment (AxeZhan) [inline pkg/scheduler/testing/wrappers.go:886]:
```suggestion
// and injects into the pv.
```

File: test/integration/scheduler/filters/filters_test.go [hunk 4]
```diff
@@ -2036,6 +2037,111 @@ func TestUnschedulablePodBecomesSchedulable(t *testing.T) {
...
+			init: func(cs kubernetes.Interface, ns string) error {
+				storage := v1.VolumeResourceRequirements{Requests: v1.ResourceList{v1.ResourceStorage: resource.MustParse("1Mi")}}
+				volType := v1.HostPathDirectoryOrCreate
+				pv, err := testutils.CreatePV(cs, st.MakePersistentVolume().
+					Name("pv-has-non-existent-nodes").
+					AccessModes([]v1.PersistentVolumeAccessMode{v1.ReadWriteOncePod}).
+					Capacity(storage.Requests).
+					HostPathVolumeSource(&v1.HostPathVolumeSource{Path: "/tmp", Type: &volType}).
+					NodeAffinityIn("kubernetes.io/hostname", []string{"node-available", "non-existing"}). // one node exist, two don't
```
Comment (AxeZhan) [inline test/integration/scheduler/filters/filters_test.go:2050]:
```suggestion
					NodeAffinityIn("kubernetes.io/hostname", []string{"node-available", "non-existing"}). // one node exists, one doesn't
```

File: test/integration/scheduler/filters/filters_test.go [hunk 5]
```diff
@@ -2036,6 +2037,111 @@ func TestUnschedulablePodBecomesSchedulable(t *testing.T) {
...
+					return fmt.Errorf("cannot create node: %w", err)
+				}
+				storage := v1.VolumeResourceRequirements{Requests: v1.ResourceList{v1.ResourceStorage: resource.MustParse("1Mi")}}
+				volType := v1.HostPathDirectoryOrCreate
+				pv, err := testutils.CreatePV(cs, st.MakePersistentVolume().
+					Name("pv-foo").
+					AccessModes([]v1.PersistentVolumeAccessMode{v1.ReadWriteOncePod}).
+					Capacity(storage.Requests).
+					HostPathVolumeSource(&v1.HostPathVolumeSource{Path: "/tmp", Type: &volType}).
+					NodeAffinityIn("foo", []string{"bar"}). // one node exist, two don't
```
Comment (AxeZhan) [inline test/integration/scheduler/filters/filters_test.go:2101]:
```suggestion
					NodeAffinityIn("foo", []string{"bar"}). 
```

Think we can just delete this comment here.

**Review by AxeZhan — COMMENTED** [review]
Thanks, looks good! Only some nits about comments.

File: test/integration/scheduler/filters/filters_test.go (hunk 5 above)
Comment (chengjoey) [reply test/integration/scheduler/filters/filters_test.go]: thanks @AxeZhan , done.

**Pull request merged**